test/mitmproxy/proxy/layers/test_websocket.py

```diff
@@ -95,12 +94,13 @@ def test_upgrade(tctx):
             >> reply()
             << SendData(tctx.client, b"\x82\nhello back")
     )
-    assert flow().handshake_flow == http_flow()
-    assert len(flow().messages) == 2
-    assert flow().messages[0].content == "hello world"
-    assert flow().messages[0].from_client
-    assert flow().messages[1].content == b"hello back"
-    assert flow().messages[1].from_client is False
+    assert len(flow().websocket.messages) == 2
+    assert flow().websocket.messages[0].content == b"hello world"
+    assert flow().websocket.messages[0].from_client
+    assert flow().websocket.messages[0].is_text
+    assert flow().websocket.messages[1].content == b"hello back"
```

Comment:
>  How about a `text` property that either works, or raises when called on a Binary message?

That's what I meant above with

> If `not is_text` the `text` property should crash or something tho?

What I don't like about this is that this means an imaginary user's addon works for five weeks and then it crashes because it sees a binary WebSocket for the first time. But on the other hand that's actually great, because if you don't check `is_text` then that's your fault and with the current design it would already result in unexpected behavior.

So yes, I think a `text` property that transparently handles the `encode`/`decode` and crashes when `not is_text` (both when reading and writing) sounds good to me.

Is it currently possible to change `is_text` (or the opcode rather, if this property is removed again)? I think it should be and the user instead should drop the message and inject a new one. Because convert from one to the other is not possible in all cases.